### PR TITLE
lint: enable nolintlinter, disable on tests

### PR DESCRIPTION
**Pull request by marbar3778**
## Description
- enable nolintlint
- disable linting on tests

Closes: #XXX

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -38,6 +41,7 @@ linters:
     # - whitespace
     # - wsl
     # - gocognit
+    # - nolintlint
```
Comment (marbar3778) [inline .golangci.yml:44]: the reviewdog docker image of golangci-lint hasnt been updated to the latest version so commenting it out for now.

File: .golangci.yml [hunk 2]
```diff
@@ -46,6 +50,7 @@ issues:
     - linters:
         - lll
       source: "https://"
+  max-same-issues: 10000
```
Comment (melekes) [inline .golangci.yml:53]: doesn't that seem high to you?

File: .golangci.yml [hunk 3]
```diff
@@ -1,3 +1,6 @@
+run:
+  tests: false
```
Comment (melekes) [inline .golangci.yml:2]: does this mean we're disabling it for tests?

Comment (marbar3778) [reply .golangci.yml:2]: yes disabling linting for tests. Do we want linting for tests?

Comment (melekes) [reply .golangci.yml:2]: Tests are important. Maybe even more so than the actual code

**Pull request merged**